### DISCUSSION TITLE
Question about removing login when Docker-hosted for node.js

### DISCUSSION THREAD

**Discussion by jbarr** [TiddlyWiki/TiddlyWiki5#7304] [Q&A] [answered]
First off, I am SO excited...again...about TiddlyWiki! I started using TW back in 2004-ish (I also ran the long gone TiddlyWikiTips site.) But I haven't used TW in a while, primarily because for the longest time, hosting TW online was not an easy task, specifically with saving changes. So fast forward to yesterday, and I discovered that you can deploy a remotely editable version of TW using Docker. Wow! Just WOW! This is a huge game changer for me. I'm now so excited to revisit my old best friend! 

Anyway, my situation is this: I'd like to remove the username/password popup login on my Docker-hosted TiddlyWiki. 

Now before you say, "No! You don't want to do that when you host in the Internet!" understand that I'm self-hosting TW on a homelab, and I'm leveraging Cloudflare. I'm using a Cloudflare Tunnel to provide remote access and a Cloudflare Application to provide authentication. My thinking is that as long as Cloudflare does its job (which, while there may be some risk, I think is a solid solution) Internet-based access should be granted to only those passing the Cloudflare Application authentication. Once authenticated, at that point, do I really need to login using a username and password in TiddlyWiki? 

So my question is, how do I remove the username/password popup on a Docker-deployed TiddlyWiki using node.js?

Thanks!

**Comment by pmario** [comment]
Hi and Welcome Back!

Which docker image do you use? .. Docker-file? .. How do you start your image? ... I know you konw it .. but anyway -- Do NOT post passwords or usernames pls ;)

**Reply by pmario** [reply]
>So my question is, how do I remove the username/password popup on a Docker-deployed TiddlyWiki using node.js?

It should not be there, except you defined it that way. Hence my questions

**Comment by jbarr** [comment]
The image is `m0wer/tiddlywiki` and I am issuing this command from a Linux host:

```
docker run -d --name=tiddlywiki \
-p 8080:8080 \
-e USERNAME=xxxx \
-e PASSWORD=xxxx \
-v /volume1/docker/tiddlywiki:/var/lib/tiddlywiki \
--restart always \
m0wer/tiddlywiki
```

Ideally, I'd like to us a Docker-Compose file to launch this using a Stack in Portainer.

Also, what image I continue to use is pretty much immaterial--I just want a trusted, stable, supported image.

**Reply by pmario** [reply]
If you remove the 2 `-e` params you should be fine

**Reply by jbarr** [reply]
> If you remove the 2 `-e` params you should be fine

That worked perfectly. The Cloudflare Application authenticates, and the Cloudflare Tunnel provides the remote access. Easy peasy.

**Comment by pmario** [comment, marked as answer]
There has been a discussion at talk.tiddlywiki.org lately: https://talk.tiddlywiki.org/t/how-to-upgrade-tw-inside-of-docker-container/6317 which uses this container https://hub.docker.com/r/nicolaw/tiddlywiki

It has 500k+ downloads .. I did _not_ test it. Just saw it's been used from the post at Talk. There may be some inspiration in the whole build system there. The GH repo is linked at DockerHub

**Reply by jbarr** [reply]
I'm marking this as the answer since I can't mark a reply to a reply as an answer, but the real answer you provided is to remote the 2 environment params. That solved it.

**Comment by Jermolene** [comment]
Hi @jbarr welcome back, and delighted you've got things up and running. We'll be interested in your feedback.

**Reply by jbarr** [reply]
So far, deploying TiddlyWiki with node.js in Docker is absolutely stellar!

Proxmox Cluster > Ubuntu VM > Docker > Portainer > TiddlyWiki Stack. Then Cloudflare Tunnel for remote access and Cloudflare Application for authentication. 

On a personal note, I'm so glad to see your continued involvement with TW. You've traveled a long road, and TW is now evolved into a mature, flexible tool. You've come a long way from 2004! And yet in its current state, though much broader in scope, TW maintains its  original speed and elegance.

**Reply by Jermolene** [reply]
Thanks @jbarr! It's great to hear people's reaction when they come back to TiddlyWiki after a time away, I hope you continue to enjoy it, and will look forwards to hearing more from you.
